Feature swc 63 crear endpoint para recibir informacion de una partida especifica

MatiasScantamburlo opened the pull request:
Listar una partida especifica por id **terminado** ¡Espero su revisión! 💯

MatiasScantamburlo committed:
feature: Agrega endpoint para obtener un lobby específico y su test

- Se agrega el endpoint GET /api/lobby/{id} en main.py para obtener la
    información de un lobby específico.
- Contiene una validación de lobby no encontrado con respuesta 404.
- También se agrega el su correspondiente test en test/test_get_game para
    verificar que se obtenga correctamente la información del lobby con el
    código de respuesta 200.

 En la rama feature_SWC-63_Crear_endpoint_para_recibir_informacion_de_una_partida_especifica
 Cambios a ser confirmados:
	modificados:     main.py
	modificados:     repositories/game.py
	nuevos archivos: tests/test_get_game.py

MatiasScantamburlo committed:
test: Se agrega test get_game_not_found

- Se implementa test para verificar que se retorne correctamente un
    mensaje 404 "Lobby not found" en caso de no existir lobby con el id
    solicitado.

 Cambios a ser confirmados:
	modificados:     tests/test_get_game.py

KaiserOfNone commented on main.py:
```diff
@@ -30,3 +31,10 @@ def get_games_repo(request: Request) -> GameRepository:
+
+@app.get("/api/lobby/{id}")
+def get_game(id: int, repo: GameRepository = Depends(get_games_repo)):
+    lobby = repo.get(id)
```
No, nunca mandes lo que tenemos en la DB pelado al cliente. En la DB estan los identificadoes de cada jugador que son secretos.
Arma una class `GameStateOutput` o algo asi que siga lo que dice en el ticket.

MatiasScantamburlo replied:
Sisi, tenés razón ¡Gracias Andi!

KaiserOfNone commented on repositories/game.py:
```diff
@@ -20,7 +20,7 @@ def get(self, id: int) -> Optional[Game]:
         return self.db.query(Game).limit(count).all()
 
-    def get_available(self, count: int = None) -> List[Game]:
+    def get_available(self, count: int = 10) -> List[Game]:
```
Porque cambiaste esto? 
1. No va al ticket.
2. El None te permite no pasar el argumento y que te de todos. Este cambio elimina ese comportamiento.

MatiasScantamburlo replied:
Mala mía, se me filtro en el commit. Obviamente no va

No te preocupes entiendo el funcionamiento, solo que el editor me lo marcaba como error de tipado y lo cambie para que no me dolieran los ojos jajaja.

KaiserOfNone replied:
Podes cambiarlo por un `Optional[int]` si apacigua al linter.

MatiasScantamburlo replied:
Me encantó. Usé:
`get_available(self, count: int | None = None)`
supongo que es lo mismo ¿no?

KaiserOfNone replied:
Si es onda lo mismo.

KaiserOfNone commented on tests/test_get_game.py:
```diff
@@ -0,0 +1,31 @@
+    "min_players": 2
+}
+
+def test_get_game():
```
Agrega jugadores a la partida en el test.

MatiasScantamburlo committed:
Crea clase GameStateOutput y mapea la salida de los endpoints

- Añadida la clase GameStateOutput para estructurar la salida de los endpoints
    /api/lobby y /api/lobby/{id}.
- Mapeada la información de las partidas disponibles y de un lobby específico
    a la nueva estructura.
- La lógica implementada devolvera detalles como el nombre de la partida,
    número de jugadores actuales, jugadores máximos/mínimos, estado de la
    partida e información sobre el turno actual.

 Cambios a ser confirmados:
	modificados:     main.py
	modificados:     repositories/game.py

MatiasScantamburlo committed:
Actualiza data_out en los tests para usar una instancia de Game

- Modificado data_out en los tests, pasando de un diccionario simple a una
    instancia de Game.
- La nueva instancia de Game incluye una lista de jugadores, mejorando la
    precisión en las pruebas al reflejar la estructura actual de los modelos.

 Cambios a ser confirmados:
	modificados:     tests/test_get_game.py

KaiserOfNone commented on main.py:
```diff
@@ -15,6 +16,14 @@
 app = FastAPI()
 game_repo = GameRepository(db.session())
 
+class GameStateOutput(BaseModel):
```
Veo que no hay data sobre los jugadores. No importa por ahora, se puede hacer despues. Escribi un ticket de "agregar informacion de tarjetas" o algo asi para ponerlo despues. (dado que no tenemos mucho de tarjetas aun)

Este comment no es bloqueante, es informativo.

KaiserOfNone reviewed (approved):
Corre `make format` y podes mergearlo.

MatiasScantamburlo committed:
Arreglamos formato

MatiasScantamburlo committed:
Merge branch 'develop' into feature_SWC-63_Crear_endpoint_para_recibir_informacion_de_una_partida_especifica
